src/Analyzers/CSharp/Tests/RemoveUnusedParametersAndValues/RemoveUnusedValueAssignmentTests.cs

```diff
@@ -318,6 +318,67 @@ void M()
             }.RunAsync();
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnusedValues)]
+        public async Task Assignment_ConstantValue_NoReads_TopLevel_DiscardOption()
+        {
+            var source = @"
+int x;
+{|IDE0059:x|} = 1;
+
+System.Console.WriteLine();
+";
+            // The code fix offered doesn't make much sense. The variable 'x' is still there.
+            var fixedSource = @"
+int x;
+_ = 1;
+
+System.Console.WriteLine();
+";
```

Comment:
- Should we use a discard here? While `CSharpCodeStyleOptions.UnusedValueAssignment` is set to `DiscardVariable`, but above test that use `DiscardVariable` doesn't actually use a discard. I assume it only uses a discard for non-constants (e.g method calls)?

- If we should use a discard, then `int x;` should be removed.

- Writing this code in the IDE shows there is IDE0059 on both lines `int x;` and `x = 1;`. I'm surprised the test passes with only a single line containing the diagnostic span. Is the behavior of the current master is actually different than what I see in the IDE? Or there is another reason for the test to pass?

Reply:
I agree that it should be removed if a discard is kept. But, should it not try to match the behavior of the remove unused value cases that are not top level? What would be the disadvantage of just removing the line entirely?